### ISSUE TITLE
Remove date-formatting restriction in filenames

### ISSUE THREAD

**Issue by gwgundersen** [gwgundersen/anno#14]
Anno expects filenames to be in the format `YYYY-MM-DD-filename.anno.md`. This is nice for me, but does not generalize to how other people may naturally think when using the software. See user errors such as #13. Just remove this restriction. Maybe add a flag or a formatting option so that users can restrict filenames to a particular format if they like.

**Comment by ysr23** [comment]
with further digging around, i found, that while the filenames were wrong, that it was the lack of yaml at the top of the .md file that was causing the error

**Comment by gwgundersen** [comment]
I see. I can also look into a better error message in this case.

**Comment by gwgundersen** [comment]
Closed by #17 by allowing the user to specify `gen_fname_base` in `.anno_config.py`:

```python
def gen_fname_base(fm):
    """`fm` is a dictionary of key-value pairs representing data in the
    notes' frontmatter.
    """
    # Examples:
    # return fm['title']
    # return np.random.randint(1e6)
    # return f"{fm['date']}-{fm['title']}"
    pass

c['gen_fname_base'] = gen_fname_base
```
